src/library/scala/collection/mutable/ArrayBuffer.scala

```diff
@@ -245,7 +250,18 @@ object RefArrayUtils {
       Array.copy(array, 0, newArray, 0, end)
       newArray
     }
-    if (n <= arrayLength) array else growArray
+    def shrinkArray = {
+      var newSize: Long = math.max(arrayLength * 2, 8)
+      while (newSize / 2 > n && newSize > 8)
+        newSize = newSize / 2
+      val newArray: Array[AnyRef] = new Array(newSize.toInt)
+      Array.copy(array, 0, newArray, 0, end)
+      newArray
+    }
+
+    if (n > arrayLength) growArray
+    else if (n <= arrayLength / 4) shrinkArray
```

Comment:
could be `else if (arrayLength > 8 && n <= arrayLength / 4)`

Reply:
I have extracted a `targetSize` variable, which is the maximum between 8 and the length `n` times 2.